JsonNumberFormat seems Locale dependent

**Describe the bug**
Currently when I try to build yasson, I get the following build error over here in Europe/Austria:

```
[ERROR] Failures:
[ERROR]   DocumentationExampleTest.testDateNumberFormats1:365 expected: <{"birthDate":"07.08.1999","name":"Jason Bourne","salary":"123.46"}> but was: <{"birthDate":"07.08.1999","name":"Jason Bourne","salary":"123,46"}>
[ERROR]   DocumentationExampleTest.testDateNumberFormats2:395 expected: <123.45678> but was: <123>
```
This is because in Austria our decimal separator is not a dot but a comma. But this is likely a bad idea to have a test for a JSON representation Locale dependent.

**To Reproduce**
Switch your system Locale to Austria and run a build

**Expected behavior**
Builds should pass, regardless where on earth they are executed.

Just define the locale = "US_en" in the test pojo.

**System information:**
 - OS: MacOS
 - Java Version: 17
 - Yasson Version: latest checkout from git 3.0.5-SNAPSHOT from today

--- a/.github/workflows/maven.yml
+++ b/.github/workflows/maven.yml
@@ -32,7 +32,7 @@ jobs:
         java_version: [ 11, 17, 21 ]
         include:
           - locale: en_US
-          # Use a different locale when running against once of the java versions
+          # Use a different locale when running against one of the java versions
           # to ensure we correctly write tests that account for localization differences.
           - locale: de_AT
             java_version: 21

--- a/.github/workflows/maven.yml
+++ b/.github/workflows/maven.yml
@@ -24,12 +24,17 @@ on:
 
 jobs:
   build:
-    name: Test on JDK ${{ matrix.java_version }}
+    name: Test on JDK ${{ matrix.java_version }} locale ${{ matrix.locale }}
     runs-on: ubuntu-latest
 
     strategy:
       matrix:
         java_version: [ 11, 17, 21 ]
+        locale: [ en_US, en_GB ]
+
+    env:
+      LANG: ${{ matrix.locale }}.UTF-8
+      LC_ALL: ${{ matrix.locale }}.UTF-8
 
     steps:
       - name: Checkout for build
@@ -50,8 +55,5 @@ jobs:
         run: mvn -U -C clean install -DskipTests
       - name: Yasson tests
         run: mvn -U -B -C -Dmaven.javadoc.skip=true verify
-      # Display numbers/dates using a different locale to ensure that the internationalization code is working properly
-      - name: Yasson international tests
-        run: mvn -U -B -C -Dmaven.javadoc.skip=true -Duser.language=en -Duser.country=GB verify
       - name: JSONB-API TCK
         run: cd yasson-tck && mvn -U -B test -DargLine="-Djava.locale.providers=COMPAT"

--- a/.github/workflows/maven.yml
+++ b/.github/workflows/maven.yml
@@ -30,11 +30,12 @@ jobs:
     strategy:
       matrix:
         java_version: [ 11, 17, 21 ]
-        locale: [ en_US, en_GB ]
-
-    env:
-      LANG: ${{ matrix.locale }}.UTF-8
-      LC_ALL: ${{ matrix.locale }}.UTF-8
+        include:
+          - locale: en_US
+          # Use a different locale when running against once of the java versions
+          # to ensure we correctly write tests that account for localization differences.
+          - locale: de_AT
+            java_version: 21
 
     steps:
       - name: Checkout for build
@@ -54,6 +55,12 @@ jobs:
       - name: Yasson install
         run: mvn -U -C clean install -DskipTests
       - name: Yasson tests
-        run: mvn -U -B -C -Dmaven.javadoc.skip=true verify
+        run: |
+          export LANG_TAG="${{ matrix.locale }}"
+          mvn -U -B -C \
+            -Dmaven.javadoc.skip=true \
+            -Duser.language="${LANG_TAG%%_*}" \
+            -Duser.country="${LANG_TAG##*_}" \
+            verify
       - name: JSONB-API TCK
         run: cd yasson-tck && mvn -U -B test -DargLine="-Djava.locale.providers=COMPAT"

--- a/.github/workflows/maven.yml
+++ b/.github/workflows/maven.yml
@@ -50,5 +50,8 @@ jobs:
         run: mvn -U -C clean install -DskipTests
       - name: Yasson tests
         run: mvn -U -B -C -Dmaven.javadoc.skip=true verify
+      # Display numbers/dates using a different locale to ensure that the internationalization code is working properly
+      - name: Yasson international tests
+        run: mvn -U -B -C -Dmaven.javadoc.skip=true -Duser.language=en -Duser.country=GB verify
       - name: JSONB-API TCK
         run: cd yasson-tck && mvn -U -B test -DargLine="-Djava.locale.providers=COMPAT"

--- a/.gitignore
+++ b/.gitignore
@@ -4,7 +4,7 @@
 .project
 .idea/
 .settings/
-/.DS_Store
+.DS_Store
 bin/
 .envrc
 .vscode/

--- a/src/test/java/org/eclipse/yasson/documented/DocumentationExampleTest.java
+++ b/src/test/java/org/eclipse/yasson/documented/DocumentationExampleTest.java
@@ -351,7 +351,7 @@ public class DocumentationExampleTest {
         @JsonbDateFormat("dd.MM.yyyy")
         public LocalDate birthDate;
 
-        @JsonbNumberFormat("#0.00")
+        @JsonbNumberFormat(value = "#0.00")
         public BigDecimal salary;
     }
     
@@ -375,6 +375,7 @@ public class DocumentationExampleTest {
 
         public LocalDate birthDate;
 
+        @JsonbNumberFormat(value = "#0.00") // TODO: remove if withNumberFormat is added to JsonbConfig builder
         public BigDecimal salary;
     }
     
@@ -385,14 +386,14 @@ public class DocumentationExampleTest {
         p.birthDate = LocalDate.of(1999, 8, 7);
         p.salary = new BigDecimal("123.45678");
         Jsonb jsonb = JsonbBuilder.create(new JsonbConfig()//
-                .withDateFormat("dd.MM.yyyy", null)); // TODO: why no withNumberFormat?
+                .withDateFormat("dd.MM.yyyy", null)); // TODO: add withNumberFormat if added to JsonbConfig builder
         String json = jsonb.toJson(p);
-        assertEquals("{\"birthDate\":\"07.08.1999\",\"name\":\"Jason Bourne\",\"salary\":123.45678}", json);
+        assertEquals("{\"birthDate\":\"07.08.1999\",\"name\":\"Jason Bourne\",\"salary\":123.46}", json);
         
-        Person9 after = jsonb.fromJson("{\"birthDate\":\"07.08.1999\",\"name\":\"Jason Bourne\",\"salary\":123.45678}", Person9.class);
+        Person9 after = jsonb.fromJson("{\"birthDate\":\"07.08.1999\",\"name\":\"Jason Bourne\",\"salary\":123.46}", Person9.class);
         assertEquals(p.name, after.name);
         assertEquals(p.birthDate, after.birthDate);
-        assertEquals(p.salary, after.salary);
+        assertEquals(new BigDecimal("123.46"), after.salary);
     }
     
     public static class Customer {

--- a/src/test/java/org/eclipse/yasson/documented/DocumentationExampleTest.java
+++ b/src/test/java/org/eclipse/yasson/documented/DocumentationExampleTest.java
@@ -351,7 +351,7 @@ public class DocumentationExampleTest {
         @JsonbDateFormat("dd.MM.yyyy")
         public LocalDate birthDate;
 
-        @JsonbNumberFormat(value = "#0.00")
+        @JsonbNumberFormat(value = "#0.00", locale="en_US")
         public BigDecimal salary;
     }
     
@@ -375,11 +375,11 @@ public class DocumentationExampleTest {
 
         public LocalDate birthDate;
 
-        @JsonbNumberFormat(value = "#0.00") // TODO: remove if withNumberFormat is added to JsonbConfig builder
+        @JsonbNumberFormat(value = "#0.00", locale="en_US") // TODO: remove if withNumberFormat is added to JsonbConfig builder
         public BigDecimal salary;
     }
     
-    @Test
+    @Test //TODO https://github.com/eclipse-ee4j/yasson/issues/722
     public void testDateNumberFormats2() {
         Person10 p = new Person10();
         p.name = "Jason Bourne";
@@ -388,9 +388,9 @@ public class DocumentationExampleTest {
         Jsonb jsonb = JsonbBuilder.create(new JsonbConfig()//
                 .withDateFormat("dd.MM.yyyy", null)); // TODO: add withNumberFormat if added to JsonbConfig builder
         String json = jsonb.toJson(p);
-        assertEquals("{\"birthDate\":\"07.08.1999\",\"name\":\"Jason Bourne\",\"salary\":123.46}", json);
+        assertEquals("{\"birthDate\":\"07.08.1999\",\"name\":\"Jason Bourne\",\"salary\":\"123.46\"}", json);
         
-        Person9 after = jsonb.fromJson("{\"birthDate\":\"07.08.1999\",\"name\":\"Jason Bourne\",\"salary\":123.46}", Person9.class);
+        Person9 after = jsonb.fromJson("{\"birthDate\":\"07.08.1999\",\"name\":\"Jason Bourne\",\"salary\":\"123.46\"}", Person9.class);
         assertEquals(p.name, after.name);
         assertEquals(p.birthDate, after.birthDate);
         assertEquals(new BigDecimal("123.46"), after.salary);

